app/models/main.py

```diff
@@ -1832,6 +1837,276 @@ def serialize(self):
         }
 
 
+class ProcessOutcome(db.Model):
+    __tablename__ = "process_outcomes"
+
+    RESULT_CHOICES = (
+        "awarded",
+        "cancelled",
+        "none-suitable",
+    )
+
+    id = db.Column(db.Integer, primary_key=True)
+    external_id = db.Column(db.BigInteger, nullable=False, default=random_positive_external_id, unique=True)
+    completed_at = db.Column(db.DateTime, nullable=True)
+
+    start_date = db.Column(db.Date, nullable=True)
+    end_date = db.Column(db.Date, nullable=True)
+    awarding_organisation_name = db.Column(db.String, nullable=True)
+    # should be enough scale to represent up to £1b - 1p
+    award_value = db.Column(db.Numeric(11, 2), nullable=True)
+
+    result = db.Column(db.String, nullable=False)
+
+    direct_award_project_id = db.Column(
+        db.Integer,
+        db.ForeignKey('direct_award_projects.id'),
+        nullable=True,
+    )
+    direct_award_search_id = db.Column(
+        db.Integer,
+        db.ForeignKey('direct_award_searches.id'),
+        nullable=True,
+    )
+    direct_award_archived_service_id = db.Column(
+        db.Integer,
+        db.ForeignKey('archived_services.id'),
+        nullable=True,
+    )
+
+    brief_id = db.Column(
+        db.Integer,
+        db.ForeignKey('briefs.id'),
+        nullable=True,
+    )
+    brief_response_id = db.Column(
+        db.Integer,
+        db.ForeignKey('brief_responses.id'),
+        nullable=True,
+    )
+
+    # a concise wrapper to reduce the ugliness of the workaround for https://bitbucket.org/zzzeek/sqlalchemy/issues/3999
+    # (which should no longer be needed once we switch to sqlalchemy 1.2)
+    def c(expression):
+        return db.cast(expression, db.Boolean)
+
+    __table_args__ = (
+        db.CheckConstraint(sql_case(
+            (
+                # if awarded, the "null-ness" of direct_award_project_id, direct_award_search_id &
+                # direct_award_archived_service_id must be the same
```

Comment:
In what situation should `result == awarded` and these all be null?

Reply:
Hmmmmmm quite possibly.